src/main/java/com/minecolonies/coremod/entity/ai/citizen/cook/EntityAIWorkCook.java

```diff
@@ -346,46 +448,105 @@ private AIState cookFood()
             if (entity instanceof TileEntityFurnace && !((TileEntityFurnace) entity).isBurning()
                     && (!ItemStackUtils.isEmpty(((TileEntityFurnace) entity)
                     .getStackInSlot(RESULT_SLOT))
                     || !ItemStackUtils.isEmpty(((TileEntityFurnace) entity).getStackInSlot(COOK_SLOT))))
             {
-                worker.setLatestStatus(new TextComponentTranslation("com.minecolonies.coremod.status.retrieving"));
                 return pos;
             }
         }
         return null;
     }
 
+    /**
+     * Specify that we dump inventory after every action.
+     * @see com.minecolonies.coremod.entity.ai.basic.AbstractEntityAIBasic#getActionsDoneUntilDumping()
+     * @return 1 to indicate that we dump inventory after every action
+     */
     @Override
     protected int getActionsDoneUntilDumping()
     {
         return 1;
     }
 
+    /**
+     * Execute tasks in this order: Retrieve food or cook food in furnaces, request food, serve food, cook food, other tasks.
+     *
+     * If any furnace has food waiting to be cooked, Find the first furnace in the building which is not cooking and either has a cooked result or has an uncooked input
+     * and walk to that furnace's position by transitioning to COOK_RETRIEVE_FOOD.
+     * If any furnace has food waiting to be cooked, and if no such furnace, then COOK_COOK_FOOD.
+     * If no food in the building nor inventory, set status to "Waiting For Food" and request food if we haven't already done so. In any case, repeat this state.
+     * If we have food in building or inventory, find all citizens nearby that are not cooks and have a zero or negative saturation.
+     * If there is a hungry citizen and we have food in inventory, then COOK_SERVE
+     * If there is a hungry citizen and we have no food in inventory, then COOK_GATHER_FOOD_FROM_BUILDING
+     * if no hungry citizens and we have less than 64 * building-level food stores, request food if we haven't already done so.
+     * If there is uncooked food in building or inventory, then COOK_COOK_FOOD.
+     * If any furnace is idle and has cooked food, then COOK_RETRIEVE_FOOD.
+     * If any furnace is idle and has uncooked food, then COOK_COOK_FOOD.
+     * Otherwise, set status to "Idling", then delay and START_WORKING
+     *
+     * @return next AIState
+     */
     private AIState startWorking()
     {
-        if (((BuildingCook) getOwnBuilding()).isSomethingInOven(world))
+        worker.setLatestStatus(new TextComponentTranslation(TranslationConstants.COM_MINECOLONIES_COREMOD_STATUS_DECIDING));
+
+        // TODO: Doesn't seem like checking the ovens should be our first order of business.
+        // TODO: seems like we should try to serve food if we can, then cook if we can, then request food if no other options.
+        // TODO: If we have hungry customers, we might want to request food even if we could possibly cook it as well.
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)